controllers/workspace/provision/deployment.go

```diff
@@ -107,6 +109,35 @@ func SyncDeploymentToCluster(
 		}
 	}
 
+	// Get pod and update dw status on crashloopbackoff
+	podList, err := getPods(workspace, clusterAPI.Client)
+	if err != nil {
+		return DeploymentProvisioningStatus{
+			ProvisioningStatus: ProvisioningStatus{
+				Requeue: true,
+				Err:     err,
+			},
+		}
+	}
+
+	for _, pod := range podList.Items {
+		if pod.Status.Phase == corev1.PodPending {
+			return DeploymentProvisioningStatus{
+				ProvisioningStatus: ProvisioningStatus{Requeue: true},
```

Comment:
I don't know if we need to requeue here. If a pod is pending, either it will still be pending (status hasn't changed and the requeue has no effect) or the phase will have changed (which would trigger a new reconcile anyways).

Reply:
Yeah, that's a good point, i've removed it!